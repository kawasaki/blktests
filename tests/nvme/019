#!/bin/bash
# SPDX-License-Identifier: GPL-2.0+
# Copyright (c) 2017-2018 Western Digital Corporation or its affiliates.
#
# Test NVMe DSM Discard command on NVMeOF with a block-device ns.

. tests/nvme/rc

DESCRIPTION="test NVMe DSM Discard command on NVMeOF block-device ns"
QUICK=1

requires() {
	_nvme_requires
	_have_loop
	_require_nvme_trtype_is_fabrics
}

test() {
	echo "Running ${TEST_NAME}"

	_setup_nvmet

	local nvmedev
	local nblk_range="10,10,10,10,10,10,10,10,10,10"
	local sblk_range="100,200,300,400,500,600,700,800,900,1000"

	_nvmet_target_setup

	_nvme_connect_subsys "${nvme_trtype}" "${def_subsysnqn}"

	nvmedev=$(_find_nvme_dev "${def_subsysnqn}")
	_check_uuid "${nvmedev}"

	nvme dsm "/dev/${nvmedev}" -n 1 -d -s "${sblk_range}" -b "${nblk_range}"

	_nvme_disconnect_subsys "${def_subsysnqn}"

	_nvmet_target_cleanup

	echo "Test complete"
}
