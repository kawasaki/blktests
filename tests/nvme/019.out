Running nvme/019
NVMe DSM: success
NQN:blktests-subsystem-1 disconnected 1 controller(s)
Test complete
